### PR TITLE
Error middleware

### LINKED ISSUE #23: Error Middleware
Middleware for handling all type of errors

### PR DESCRIPTION
**Pull request by Ibtisam-Hemmo**

### CODE REVIEW

File: server/validation/signupSchema.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    fullName:string,
+    ownerID: number,
+    pharmacyName:string,
+    licenseNumber:number,
+    pharmacyLocation:string,
+    phoneNumber:number,
+    email:string,
+    password:string,
+    confirmPassword:string
+}
```
Comment (Raghad-Mezied) [inline server/validation/signupSchema.ts]: Move to interfaces

File: server/utils/generateToken.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+require('env2')('.env');
+
+type payloadType = {
+    fullName:string,
+    ownerID: number,
+    ownerImg:string
+    pharmacyName:string,
+}
+
```
Comment (Raghad-Mezied) [inline server/utils/generateToken.ts]: Move to interfaces

File: server/utils/generateToken.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { sign } from 'jsonwebtoken';
+
+require('env2')('.env');
```
Comment (Raghad-Mezied) [inline server/utils/generateToken.ts]: No need to require here as you did in your server file

File: server/middlewares/Error.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+const { status, message } = CustomError;
+
+const ErrorMiddleware = ((err: CustomError, req: Request, res: Response, next: NextFunction) => {
+  if (!status) {
+    res.status(500).json({ msg: message });
+  } else {
+    res.status(status).json({ msg: message });
+  }
+});
```
Comment (Raghad-Mezied) [inline server/middlewares/Error.ts]: The status and the message are from your error, not the custom error

File: server/controllers/authentication/signUp.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+import { hash } from 'bcryptjs';
+import Pharmacy from '../../models/pharmacy';
+import signupSchema from '../../validation/signupSchema';
+import generateToken from '../../utils/generateToken';
+
+const signUp = async (req: Request, res: Response) => {
+  try {
+    await signupSchema(req.body);
+
+    const licenseExisted = await Pharmacy.findOne({
```
Comment (Raghad-Mezied) [inline server/controllers/authentication/signUp.ts]: Delete the signup part.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Review by AhmedAbuSharar — APPROVED** [review]

**Review by Nader-SH — APPROVED** [review]

**Pull request merged**